### PR TITLE
Kc set up recommended

### PR DESCRIPTION
**Pull request by kev1n80**
+ Abstract the `filterQuery` method and move it from out of the FetchListings file and into its own filter folder and file
+ Move FetchListings constants into ListingContants file
+ Change the sort by input values from Strings to ints
  - This makes the request smaller and it translates better into different languages
+ Fix the `isErrorMessage` function
+ Prep the sortBy algorithms by creating a sort folder and if statements to call each sorting algorithm
  - Prep the recommended sorting algorithm
      + It will call the `sortByRecommended` function
        - Which will call the `sortByUpvotedListings` (Kevin's) function if a User is logged in and has an account
        - Which will call the `sortByRepAndLocation` (Jaden's) function regardless if a User is logged in or not
        - Neither of these functions have been created yet

### CODE REVIEW

File: src/main/java/com/google/sps/sort/Recommended.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+   * @return a List<Listing> based on a User's upvoted listings or the listings 
+   *     reputation and location.
+   */
+  public static List<Listing> sortByRecommended(DatastoreService datastore, 
+      List<Listing> listings, UserService userService) {
+    List<Listing> sortedListings = new ArrayList<Listing>();
+    // If the user is logged in then sort by upvoted listings first
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      if (AuthenticationUtility.userAlreadyHasAccount(datastore, userEmail)) {
```
Comment (zrideaux) [inline src/main/java/com/google/sps/sort/Recommended.java:43]: I think you can safely bypass this check (line 43) -- this method is primarily to see if a new account needs to be made in the authentication servlet. My understanding is that Authentication.java will be run prior to looking for listings, you can assume they have an account already, and `isUserLoggedIn` should be a sufficient check! Let me know if I misinterpreted the usage though :)

File: src/main/java/com/google/sps/sort/Recommended.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.users.UserService;
+import com.google.sps.data.Listing;
+import com.google.sps.utility.AuthenticationUtility;
+import java.util.ArrayList;
+import java.util.List;
+
+// The alogrithm to sort Listings by Recommended
+public final class Recommended {
```
[the remark this answers is not in the document]
Comment (kev1n80) [reply src/main/java/com/google/sps/sort/Recommended.java:26]: Yup that sounds great!

File: src/main/java/com/google/sps/sort/Recommended.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  public static List<Listing> sortByUpvotedListings(List<Listing> listings, 
+      Entity userEntity) {
+    // TODO
+    return listings;
+  }
+
+  /**
+   * TODO
+   */
+  public static List<Listing> sortByRepAndLocation(List<Listing> listings) {
```
[the remark this answers is not in the document]
Comment (kev1n80) [reply src/main/java/com/google/sps/sort/Recommended.java:76]: Yup sounds great!

File: src/main/java/com/google/sps/sort/Recommended.java (hunk 1 above)
Comment (kev1n80) [reply src/main/java/com/google/sps/sort/Recommended.java:43]: Nope, that sounds great. Should I do the same thing for the FetchUser servlet? Currently it has both checks

Comment (zrideaux) [reply src/main/java/com/google/sps/sort/Recommended.java]: You should be fine to do that! If it does end up causing any problems let me know and I can look into it!

**Pull request merged**